terraso_backend/apps/auth/middleware.py

```diff
@@ -23,56 +29,82 @@
 
 
 class JWTAuthenticationMiddleware:
-    def __init__(self, get_response):
-        self.get_response = get_response
+    def process_view(self, request, view_func, view_args, view_kwargs):
+        auth_optional = getattr(view_func, "auth_optional", False)
+        auth_required = not self._is_path_public(request.path) and not auth_optional
 
-    def __call__(self, request):
-        if not request.user or not request.user.is_authenticated:
-            user = self._get_user_from_jwt(request)
+        if request.user.is_authenticated or (not auth_required and not auth_optional):
+            return None
+
+        auth_header = request.META.get("HTTP_AUTHORIZATION")
+        if not auth_header and auth_optional:
+            request.user = None
```

Comment:
question: Could we replace this with [AnonymousUser](https://docs.djangoproject.com/en/4.2/topics/auth/default/#authentication-in-web-requests)? I'm still not super familiar with how our current authentication setup works, but it seems like `AnonymousUser` is what is used in Django when the user is not logged in. I'm just wondering if some Django plugins might rely on `AnonymousUser` (I don't think any that we use currently, but maybe in the future).

Reply:
I thought that was happening automatically but it seems that is not the case, I will check how to use AnonymousUser

Reply:
@david-code I updated it to use AnonymousUser